### ISSUE TITLE
Unable to get default parameters of STL function

### ISSUE THREAD

**Issue by Ashish-Tiger** [statsmodels/statsmodels#8701]
#### Describe the bug

[A clear and concise description of what the bug is. This should explain **why** the current behaviour is a problem and why the expected output is a better solution.]

#### Code Sample, a copy-pastable example if possible

```python
import statsmodels.tsa.seasonal.STL
import inspect 
inspect.signature(STL.__init__)
# <Signature (self, /, *args, **kwargs)>
```
<details>

**Note**: As you can see, there are many issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates.

**Note**: Please be sure you are using the latest released version of `statsmodels`, or a recent build of `main`. If your problem has been fixed in an unreleased version, you might be able to use `main` until a new release occurs. 

**Note**: If you are using a released version, have you verified that the bug exists in the main branch of this repository? It helps the limited resources if we know problems exist in the current main branch so that they do not need to check whether the code sample produces a bug in the next release.

</details>

If the issue has not been resolved, please file it in the issue tracker.

#### Expected Output

expect output:

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

[paste the output of ``import statsmodels.api as sm; sm.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
Python: 3.10.8.final.0
OS: Linux 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64
byteorder: little
LC_ALL: None
LANG: C.UTF-8

statsmodels
===========

Installed: 0.13.5 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/statsmodels)

Required Dependencies
=====================

cython: Not installed
numpy: 1.23.5 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/numpy)
scipy: 1.10.0 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/scipy)
pandas: 1.5.2 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/pandas)
    dateutil: 2.8.2 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/dateutil)
patsy: 0.5.3 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.6.2 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/matplotlib)
    backend: module://matplotlib_inline.backend_inline 
cvxopt: Not installed
joblib: 1.2.0 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/joblib)

Developer Tools
================

IPython: 8.8.0 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/IPython)
    jinja2: 3.1.2 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/jinja2)
sphinx: Not installed
    pygments: 2.14.0 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/pygments)
pytest: 7.1.2 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/pytest)
virtualenv: 20.17.1 (/home/user1/miniconda3/envs/ADT/lib/python3.10/site-packages/virtualenv)
</details>

**Comment by bashtage** [comment]
This is a limitation of Cython code I believe.  Not sure there is much we can do about it.  Might be possible to use bindings to populate the signature, but I think it still won't get defaults.

**Comment by Ashish-Tiger** [comment]
Yes, I read about this particular binding `binding (True / False)` option.

Compiler directives[¶](https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#compiler-directives)

Stackoverflow link:
https://stackoverflow.com/questions/46033277/how-to-introspect-a-function-defined-in-a-cython-c-extension-module

Just trying to understand if there is a way to implement this.
